File: articles/social-identity-group-life-and-moral-polarization/go/moral_exclusion_score.go
// Synthetic moral exclusion risk example.
// Educational scaffold only.
//
// Run:
//   go run go/moral_exclusion_score.go

package main

import (
	"fmt"
	"math"
)

func logistic(x float64) float64 {
	return 1.0 / (1.0 + math.Exp(-x))
}

func main() {
	threat := 1.0
	dehumanizingRhetoric := 0.8
	loyaltyPressure := 0.7
	identityStrength := 0.6
	institutionalTrust := 0.4
	crossGroupContact := 0.3

	latent := 0.35*threat +
		0.30*dehumanizingRhetoric +
		0.25*loyaltyPressure +
		0.20*identityStrength -
		0.25*institutionalTrust -
		0.20*crossGroupContact

	risk := logistic(latent)

	fmt.Printf("Synthetic moral exclusion risk: %.3f\n", risk)
}

File: articles/social-identity-group-life-and-moral-polarization/cpp/moral_exclusion_score.cpp
/*
Synthetic moral exclusion risk example.
Educational scaffold only.

Compile:
  c++ -std=c++17 cpp/moral_exclusion_score.cpp -o outputs/tables/moral_exclusion_score_cpp
Run:
  ./outputs/tables/moral_exclusion_score_cpp
*/

#include <cmath>
#include <iostream>

double logistic(double x) {
    return 1.0 / (1.0 + std::exp(-x));
}

int main() {
    double threat = 1.0;
    double dehumanizing_rhetoric = 0.8;
    double loyalty_pressure = 0.7;
    double identity_strength = 0.6;
    double institutional_trust = 0.4;
    double cross_group_contact = 0.3;

    double latent =
        0.35 * threat +
        0.30 * dehumanizing_rhetoric +
        0.25 * loyalty_pressure +
        0.20 * identity_strength -
        0.25 * institutional_trust -
        0.20 * cross_group_contact;

    double risk = logistic(latent);

    std::cout << "Synthetic moral exclusion risk: " << risk << std::endl;
    return 0;
}
